chartfx-chart/src/main/java/io/fair_acc/chartfx/renderer/spi/AbstractRendererXY.java

```diff
@@ -95,6 +98,10 @@ public void updateAxes() {
         if (yAxis instanceof CategoryAxis axis && !getDatasets().isEmpty()) {
             axis.updateCategories(getDatasets().get(0));
         }
+
+        if (xAxisBefore != xAxis || yAxisBefore != yAxis) {
```

Comment:
This may benefit from a check whether the original [x,y]Axis isn't null and decouple the setAll(..) call. 

Otherwise OK.

Reply:
Thank you for looking into it. I am afraid I am unsure what you mean. Could you provide a code snippet that I could use?

Reply:
... probably a rare (or even impossible) corner case. Will merge this and -- in the unlikely event that this causes problems -- will follow this up.